c7n/policy.py

```diff
@@ -73,8 +75,13 @@ def __init__(self, data, options):
         # We store all the policies passed in so we can refilter later
         self._all_policies = []
         for p in self.data.get('policies', []):
-            self._all_policies.append(
-                Policy(p, options, session_factory=self.test_session_factory()))
+            if 'all' in options.regions:
+                options.regions = utils.get_available_regions(p['resource'])
```

Comment:
it's a chicken and egg problem.  I can't lookup the available regions until I know the resource.  I don't know the resource until I load the policy.